src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/FilePathNormalizer.cs

```diff
@@ -19,9 +19,13 @@ internal static class FilePathNormalizer
 
     private class FilePathNormalizingComparer : IEqualityComparer<string>
     {
+        private Func<char, char> _charConverter = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
```

Comment:
This is more for my education, but why introduce a new function here / duplicate the operating system logic for each character? [FilePathComparison](https://github.com/dotnet/razor/blob/4d44c7e82d2d56e85e4d75f986cce9f93ee44828/src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/CSharp/FilePathComparison.cs) is an existing static (factory?) that returns the case sensitivity to use. Is there any value is lowering each character individually / vs lowering the entire path outright. https://github.com/rsandbach/razor/commit/d6515d77a040ed0ee04720b8352a51185f40224c

Reply:
The reason is performance. This method can be called a lot, so allocating a string and calling ToLower on it (which allocates another) just gives more for the garbage collector to clean up. If there was a `CharComparison` type I would have used it :)